core/src/main/java/com/linecorp/armeria/server/VerbSuffixPathMapping.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+final class VerbSuffixPathMapping extends AbstractPathMapping {
+    private static final Pattern VERB_PATTERN = Pattern.compile("[^/]+:([a-zA-Z0-9-_]+)$");
+
+    private final PathMapping basePathMapping;
+    private final String verb;
```

Comment:
- `colonAndVerb`?
- We could also keep both `verb` and `colonAndVerb`, so we don't have to do substring in `doWithPrefix()`.